### DISCUSSION TITLE
关于【完全缓存（DB）】模式下分流下载优化

### DISCUSSION THREAD

**Discussion by youyou-sudo** [libsgh/PanIndex#402] [Ideas]
## 场景：
4个onedrive账号分流下载，完全缓存（DB）模式
一般操作步骤是，这四个账号挂载目录一致
1. 挂载四个onedrive存储
2. 将这四个账号设定分流名称
3. 这四个全部缓存一遍
## 想法：
既然分流的前提是目录一致，那么我觉得可以调用一个账号api缓存数据，将这个账号设定为主缓存，其他账号根据第一个缓存的账号数据来创建分流的数据
如果目录内容发生改变，那么可以刷新主缓存的数据，其他分流账号后面再刷新
## 结果
- 增加了代码量和逻辑（Σ(っ°Д°;)っ
- 只调用一个账号的api，减少了api的浪费，还能减少创建分流数据的时间（不用全部账号都调用api再缓存

**Comment by libsgh** [comment]
这种分流方式只对onedrive可以，并不通用，因为onedrive可以根据路径获取，如果是其他网盘，根据文件ID获取，文件ID是不一样的。

**Reply by youyou-sudo** [reply]
> 这种分流方式只对onedrive可以，并不通用，因为onedrive可以根据路径获取，如果是其他网盘，根据文件ID获取，文件ID是不一样的。

哦！(ﾟдﾟ)！原来是这样，还以为所有网盘都是根据路径获取

**Reply by youyou-sudo** [reply]
> 这种分流方式只对onedrive可以，并不通用

那从下载的角度，`完全缓存`模式下只产生缓存时、预览和下载的api调用，我觉得可以设置分流配置其中一个账号为`完全缓存`模式用于网页文件展示，其他账号设置为`api请求`，用户访问网页时展示文件缓存，点击下载或预览的时候分流，这样就可以只缓存一个账号的文件数据
这个方案不知道对于除Onedrive外的其他网盘可以避免掉请求文件列表所产生的调用不
